src/api/getWilayaByPhoneCode.js

```diff
@@ -14,10 +14,20 @@ const getWilayaByPhoneCode = data =>
 * @returns { Object | undefined } Returns the target object, or undefined
 */
   (phoneCode, projection) => {
-    const phoneCodeVar =  typeof phoneCode === "number"
-      ? phoneCode
-      : parseInt(phoneCode.substring(1, 3), 10);
-    const wilaya = data.find(w => w.phoneCodes.includes(phoneCodeVar));
+    let phoneCodeVar;
+    if (typeof phoneCode === 'number') {
+      phoneCodeVar = phoneCode;
+    } else {
+      const isPhoneNumberBeginWithZero = phoneCode[0] === '0';
+      const substringBegin = isPhoneNumberBeginWithZero ? 1 : 0;
+      const substringEnd = isPhoneNumberBeginWithZero ? 3 : 2;
+      phoneCodeVar =
+      phoneCode.length > 3
+        ? parseInt(phoneCode.substring(substringBegin, substringEnd), 10)
+        : parseInt(phoneCode, 10);
+    }
+    const wilaya = data.find((w) => w.phoneCodes.includes(phoneCodeVar));
     return projectWilaya(wilaya, projection);
```

Comment:
I suggest that we don't handle this kind parsing for phone codes from full phone numbers in our library, as we can receive phone codes with spacings, with country code... so it could be error prone. Let's just keep it simple and only deal with : wilaya codes (as int like `34`, or `"34"` or `"030"`).

```suggestion
    if(!Number.isInteger(phoneCode)){
      return;
    }

    const parsedPhoneCode =  parseInt(phoneCode, 10);
    const wilaya = data.find(w => w.phoneCodes.includes(parsedPhoneCode));
    // eslint-disable-next-line consistent-return
    return projectWilaya(wilaya, projection);
```

So we need to remove the test cases to only take care of "only getting a wilaya by its phone code". 

Probably we should create a separate utility method that parses the wilaya code a full phone number.

Reply:
[FIXED] yes i am agree with you, we can simply add another function for getting the wilaya by the phone number